app/views/contexts/grand_calculated_summary_context.py

```diff
@@ -0,0 +1,126 @@
+from typing import Iterable, Mapping
+
+from werkzeug.datastructures import ImmutableDict
+
+from app.questionnaire.questionnaire_schema import (
+    get_calculation_block_ids_for_grand_calculated_summary,
+)
+from app.views.contexts.calculated_summary_context import CalculatedSummaryContext
+from app.views.contexts.summary.group import Group
+
+
+class GrandCalculatedSummaryContext(CalculatedSummaryContext):
+    def _build_grand_calculated_summary_section(
+        self, rendered_block: ImmutableDict
+    ) -> dict:
+        """
+        Build list of calculated summary blocks that the grand calculated summary will be adding up
+        """
+        # Type ignore: the block, group and section will all exist at this point
+        block_id: str = self.current_location.block_id  # type: ignore
+        calculated_summary_group: ImmutableDict = self._schema.get_group_for_block_id(block_id)  # type: ignore
```

Comment:
Better to have only one type ignore here?
```suggestion
        calculated_summary_group: ImmutableDict = self._schema.get_group_for_block_id(self.current_location.block_id)  # type: ignore
```

Reply:
done 👍